### PR TITLE
Refactor(eos_designs)!: Change default descriptions for connected endpoints and network ports

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Change default descriptions for connected endpoints and network ports

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

- Adding new consistent default descriptions for connected endpoints where the type is added first in upper case. Like `SERVER_<server_name>_<server_port>`
- Add templating support for the description key under connected endpoints and network ports
- Add templating support for the descriptions[] key under connected endpoints
- Add keys with default templates:
  - `default_connected_endpoints_description`
  - `default_connected_endpoints_port_channel_description`
  - `default_network_ports_description`
  - `default_network_ports__port_channel_description`
- Add `endpoint` key to `network_ports`
- Add `endpoint_port_channel` under `port_channel` for both `network_ports` and connected endpoints.
- Updated porting guide

## How to test

- Update molecule scenarios and examples
- Verified that custom description logic still works in unit tests for templates and python module
- Verified that custom description logic still works in evpn_underlay_ebgp_overlay_ebgp for python module

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md
Comment (gmuloc) [inline ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]: I suppose this is missing
```suggestion
In AVD 5.0.0 setting the `description` key under `port_channel` of the adapter will replace the full port-channel interface description.
```

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md
Comment (gmuloc) [inline ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]:
```suggestion
allowing to use new `endpoint` key under `network_ports` which is used similarly to the endpoint name under connected endpoints.
```

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md
Comment (gmuloc) [inline ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]: Not sure here
```suggestion
In AVD 5.0.0 setting the `description` key under `port_channel` of the network port will replace the full port-channel interface description.
```

**Review by gmuloc — COMMENTED** [review]
quick comments - will review the rest afterwards

File: python-avd/pyavd/_eos_designs/schema/schema_fragments/network_ports.schema.yml
Comment (gmuloc) [inline python-avd/pyavd/_eos_designs/schema/schema_fragments/network_ports.schema.yml:40]: should we do this now?

File: python-avd/pyavd/api/interface_descriptions/__init__.py
Comment (gmuloc) [inline python-avd/pyavd/api/interface_descriptions/__init__.py:215]: in the available data are we not missing extra stuff like peer_type and peer_interface?
Should the docstring be changed?

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/AVD_LAB.yml
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/AVD_LAB.yml]:
```suggestion
---
```

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md
Comment (gmuloc) [inline ansible_collections/arista/avd/docs/porting-guides/5.x.x.md:480]: Is this a valid syntax: `{adapter_description_or_endpoint}` ?

**Review by gmuloc — COMMENTED** [review]
For network ports the fact that we use endpoints instead of description make some change for the `peer` key and hence the fabric doc that could have an impact on users relying upon it in AVD 4.x though I think the change makes sense.

File: python-avd/pyavd/_eos_designs/schema/schema_fragments/network_ports.schema.yml
Comment (ClausHolbechArista) [reply python-avd/pyavd/_eos_designs/schema/schema_fragments/network_ports.schema.yml:40]: We do not have an article. Hence I am adding the TODO :)

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/docs/porting-guides/5.x.x.md:480]: yes there is a special helper field called that. It is needed to emulate the historic behavior, so making it easier to revert to 4.x style.

File: python-avd/pyavd/api/interface_descriptions/__init__.py
Comment (ClausHolbechArista) [reply python-avd/pyavd/api/interface_descriptions/__init__.py:215]: updated

**Review by gmuloc — APPROVED** [review]
LGTM once conflicts are resolved (by regenerating the files)

**Review by carlbuchmann — APPROVED** [review]
LGTM, please review updates to documentation before merging

**Pull request merged**
